metacat-main/src/main/java/com/netflix/metacat/main/services/impl/TableServiceImpl.java

```diff
@@ -548,6 +472,12 @@ public TableDto updateAndReturn(final QualifiedName name, final TableDto tableDt
             ignoreErrorsAfterUpdate = connectorTableServiceProxy.update(name, converterUtil.fromTableDto(tableDto));
         }
 
+        // we do ownership validation and enforcement only if table owner is set in the dto
+        // because if it is null, we do not update the owner in the existing metadata record
+        if (tableDto.getTableOwner().isPresent()) {
+            ownerValidationService.enforceOwnerValidation("updateTable", name, tableDto);
```

Comment:
This `TableServiceImpl` has logic applied commonly across catalogs. To start we are focusing on ownership for hive/iceberg tables, what is your plan to be able to selectively enable validation on certain catalogs?

Reply:
The enforce method takes in the Qualified name of the table. In the pandora implementation, I was planning to add a check for the two hive catalogs. In future, if we want to do this for all catalogs, we can remove that check.

Reply:
Having said that, the existing event listener implementation (when it was enabled) did not check for catalog types before checking for the user validity. I would say eventually we want to have it for all catalogs though and the number of calls for the other catalogs should be tiny compared to hive.